add verticalScrolling

raacker opened the pull request:
Fixing #5.

Seems there are some setting value which should reset after changing direction. I had to restart app when testing vertical scrolling, not hot reload.

Please take a look :) Thanks!

raacker committed:
add verticalScrolling

ariedov commented on lib/snaplist_bloc.dart:
```diff
@@ -7,6 +7,7 @@ class SnapListBloc {
   final CardSizeProvider sizeProvider;
   final SeparatorSizeProvider separatorProvider;
   final double swipeVelocity;
+  final bool isVertical;
```
We should use the `Axis` from Dart, as it has only `horizontal` and `vertical` options. That way we explicitly define that if `isVertical == false` it will not go in circles.

ariedov replied:
The `isVertical` boolean is still useful, it just should be a property.

raacker replied:
So, keep isVertical as boolean and change builder property as Axis right?

ariedov replied:
In the end, the param should be called `axis`, and the boolean will exist as `bool get isVertical => axis == Axis.vertical`.

ariedov commented on lib/snaplist_bloc.dart:
```diff
@@ -143,26 +145,41 @@ class SnapListBloc {
-    return ((distance * 100) /
-            sizeProvider(_centerItemPosition, _createBuilderData()).width)
-        .clamp(0.0, 100.0);
+    double sizeFactor = isVertical ?
```
I'd moved the `sizeProvider(_centerItemPosition, _createBuilderData())` to a different line, and then read the `width` or `height` from it's result. I believe it's simpler that way.

ariedov commented on lib/snaplist_bloc.dart:
```diff
@@ -143,26 +145,41 @@ class SnapListBloc {
-            _nextItemPosition,
-            100.0,
-          )).width;
+      if (isVertical) {
```
The result of the `sizeProvider` call may be stored in a `final cardSize`, and then the `width` or `height` might be read from there. I'll shrink this method twice in size.

ariedov commented on lib/snaplist_view.dart:
```diff
@@ -12,6 +12,7 @@ class SnapList extends StatefulWidget {
   final ScrollProgressUpdate progressUpdate;
   final PositionUpdate positionUpdate;
   final ScrollStart scrollStart;
+  final bool verticalScrolling;
```
I already wrote the `Axis` comment.

raacker replied:
Hmm well I thought using boolean is easier to users :) even I didn’t know that dart has Axis.vertical, horizontal. But I agree with explicitly declaring value. We might better to mention it in README.md and comment description then. I’ll work on that!

ariedov replied:
The Axis is used in `ListView` and part of the `dart:ui`, so it will be familiar enough and will not need additional dependencies. And I would love if you mention it in the `README.md` =)

ariedov commented:
Great work, I requested some changes which are mostly code style. Thank you very much for your help!

raacker committed:
Modify based on request
- Shrink sizeProvider call into Size class
- take Axis value directly
- Mention axis in README.md and fix some typos :)

raacker commented:
After confirm, I'll rebase the commits

ariedov commented:
That's great, there are no conflicts. Thank you very much, I really appreciate your help!
